eng/scripts/Language-Settings.ps1

```diff
@@ -6,46 +6,24 @@ $packagePattern = "*.nupkg"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/releases/latest/dotnet-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=dotnet%2F&delimiter=%2F"
 
-function Get-dotnet-PackageInfoFromRepo ($pkgPath, $serviceDirectory)
+function Get-AllPackageInfoFromRepo($serviceDirectory)
 {
-  $projDirPath = (Join-Path $pkgPath "src")
+  $allPackageProps = @()
+  $msbuildOutput = dotnet msbuild /nologo /t:GetAllPackageInfo $EngDir/service.proj /p:ServiceDirectory=$serviceDirectory
 
-  if (!(Test-Path $projDirPath))
+  foreach ($projectOutput in $msbuildOutput)
   {
-    return $null
-  }
-
-  $projectPaths = @(Resolve-Path (Join-Path $projDirPath "*.csproj"))
+    $pkgPath, $serviceDirectory, $pkgName, $pkgVersion, $sdkType, $isNewSdk = $projectOutput.Split(' ',[System.StringSplitOptions]::RemoveEmptyEntries).Trim("'")
```

Comment:
We might not need the `[System.StringSplitOptions]::RemoveEmptyEntries`

Reply:
I mostly did that to handle any extra spaces that might be present, in particular msbuild ends up putting spaces at the beginning of the line. I could handle that differently but I think this is probably a little more forgiving.